bob-apps/bin/bobc.rs

```diff
@@ -340,11 +340,11 @@ async fn prepare_put_from_pattern(
     let mut keys_names = Vec::new();
     while let Some(entry) = dir_iter.next_entry().await.unwrap() {
         let file = entry.path();
-            let name = file.to_str().unwrap().to_owned();
-            if let (false, Some(cap)) = (file.is_dir(), re_path.captures(&name)) {
-                let key = cap[1].parse().unwrap();
-                keys_names.push(KeyName { key, name });
-            }
+        let name = file.to_str().unwrap().to_owned();
+        if let (false, Some(cap)) = (file.is_dir(), re_path.captures(&name)) {
+            let key = cap[1].parse().expect(&format!("cannot parse file name: {}", &cap[1]));
```

Comment:
This is not good solution, because `format!("cannot parse file name: {}", &cap[1])` will be called every time, even when there is no error. It is better to check for error and only in that case perform message formatting

Reply:
Done: [6bb2021](https://github.com/qoollo/bob/pull/562/commits/6bb2021f0bba6aa13da46cfdd24bc71ca49a641a)